src/helper/client.cpp

```diff
@@ -481,6 +484,12 @@ bool client::run_client_init()
 
 bool client::run_request()
 {
+    if (!request_enabled_) {
```

Comment:
```suggestion
    compute_client_status();
    if (!request_enabled_) {
```

I think it might make some sense to call it here rather than rely on the previous request status.

Reply:
It has to rely on previous otherwise the status would be calculated on the fly for reach call.  This way, the computed status is only computed on request_init or config_sync

Reply:
Ah yes to keep the atomicity of the request... However this still presents a problem with the fact that `request_init` might fail before the status can be computed, I think you still need to account for that somehow.